Orange/widgets/unsupervised/owkmeans.py

```diff
@@ -243,9 +244,9 @@ def apply(self):
         self.clear_messages()
         if self.data is not None:
             if self.optimize_k:
-                self.run_optimization()
-                self.mainArea.show()
-                self.update_results()
+                if self.run_optimization():
```

Comment:
Say that `mainArea` is hidden. If optimization fails, main area won't be shown and updated. This is OK.

What if `mainArea` is currently shown and contains some data? If optimization fails, it won't be hidden, but it won't update either...